wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/solenoid/Robot.java

```diff
@@ -25,34 +37,134 @@ public class Robot extends TimedRobot {
 
   // DoubleSolenoid corresponds to a double solenoid.
+  // In this case, it's connected to channels 1 and 2 of a PH with the default CAN ID.
   private final DoubleSolenoid m_doubleSolenoid =
-      new DoubleSolenoid(PneumaticsModuleType.CTREPCM, 1, 2);
+      new DoubleSolenoid(PneumaticsModuleType.REVPH, 1, 2);
+
+  // Compressor connected to a PH with a default CAN ID
+  private final Compressor m_compressor = new Compressor(PneumaticsModuleType.REVPH);
+
+  // External analog pressure sensor
+  // product-specific voltage->pressure conversion, see product manual
+  // in this case, 250(V/5)-25
+  // the scale parameter in the AnalogPotentiometer constructor is scaled from 1 instead of 5,
+  // so if r is the raw AnalogPotentiometer output, the pressure is 250r-25
+  static final double kScale = 250;
+  static final double kOffset = -25;
+  private final AnalogPotentiometer m_pressureTransducer =
+      new AnalogPotentiometer(/* the AnalogIn port*/ 2, kScale, kOffset);
 
-  private static final int kSolenoidButton = 1;
-  private static final int kDoubleSolenoidForward = 2;
-  private static final int kDoubleSolenoidReverse = 3;
+  // A chooser to select the compressor mode from the dashboard
+  private final SendableChooser<CompressorConfigType> m_compressorModeChooser =
+      new SendableChooser<>();
+
+  static final int kSolenoidButton = 1;
+  static final int kDoubleSolenoidForwardButton = 2;
+  static final int kDoubleSolenoidReverseButton = 3;
+  static final int kCompressorButton = 4;
+
+  @Override
+  public void robotInit() {
+    // Publish elements to shuffleboard.
+    ShuffleboardTab tab = Shuffleboard.getTab("Pneumatics");
+    tab.add("Single Solenoid", m_solenoid);
+    tab.add("Double Solenoid", m_doubleSolenoid);
+    tab.add("Compressor", m_compressor);
+
+    // Also publish some raw data
+    tab.addDouble(
+        "PH Pressure [PSI]",
+        () -> {
+          // Get the pressure (in PSI) from the analog sensor connected to the PH.
+          // This function is supported only on the PH!
+          // On a PCM, this function will return 0.
+          return m_compressor.getPressure();
+        });
+    tab.addDouble(
+        "External Pressure [PSI]",
+        () -> {
+          // Get the pressure (in PSI) from an analog pressure sensor connected to the RIO.
+          return m_pressureTransducer.get();
+        });
+    tab.addDouble(
+        "Compressor Current",
+        () -> {
+          // Get compressor current draw.
+          return m_compressor.getCurrent();
+        });
+    tab.addBoolean(
+        "Pressure Switch",
+        () -> {
+          // Get the digital pressure switch connected to the PCM/PH.
+          // The switch is open when the pressure is over ~120 PSI.
+          return m_compressor.getPressureSwitchValue();
+        });
+
+    // Add the options and send the chooser to the dashboard.
+    m_compressorModeChooser.setDefaultOption("Digital", CompressorConfigType.Digital);
```

Comment:
I don't like using a sendable chooser for this, because it isn't something you'd actually want to do on a robot. You'd pick one. I'd rather just have all three in the code with 2 commented out and a comment to pick the one you want to use.